examples/03_evaluate/als_movielens_diversity_metrics.ipynb

```diff
@@ -0,0 +1,681 @@
+{
```

Comment:
<p>I really like this notebook, I think it would be good to add some final thoughts here and explain why ALS outperforms Random in ranking but not in diversity. Maybe also define diversity metrics, why we want them, and in which cases it would be beneficial to take into account diversity metrics instead of rating or ranking</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/microsoft/recommenders/pull/1416/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Reply:
addressed

Reply:
this is really good!

Reply:
I am thinking that it looks a bit as if we suggest random as a valid algorithm. I may rewrite a bit to emphasize the trade off i.e. one doesn't want maximum diversity when doing recommendations.

Reply:
haha, yeah that's definitely not the intent. i don't think it buys us much to show the typical rating / ranking performance metrics. imo it's better to focus exclusively on diversity and show how random recommendations enforce diversity which is shown in the metrics (to prove the metrics work)

Reply:
Another thing that would be nice is to explain a bit the definitions of the metrics or where they can be found. It is not clear to the reader in which reference each definition appears.

Reply:
sounds good, do you mind capturing these items in a new issue focused on diversity metric improvements since this pr is already merged?

Reply:
https://github.com/microsoft/recommenders/issues/1453